File: TAREA 1-7/Tarea No. 4/PYTHON/suma_de_los_numero_imapres_en_la_lista.py


# Dados diez números enteros, visualizar la suma de los números pares de la lista,
#cuántos números pares existen y cuál es la media aritmética de los números impares

numeros = []
suma_pares = 0
conteo_pares = 0
suma_impares = 0
conteo_impares = 0

# Leer 10 números enteros del usuario
print("Introduce 10 numeros enteros:")
for _ in range(10):
    numero = int(input())
    numeros.append(numero)

# Procesar los números para encontrar pares e impares
for numero in numeros:
    if numero % 2 == 0:  # Verifica si el número es par
        suma_pares += numero
        conteo_pares += 1
    else:  # El número es impar
        suma_impares += numero
        conteo_impares += 1

# Mostrar resultados
print(f"Suma de los numeros pares: {suma_pares}")
print(f"Cantidad de numeros pares: {conteo_pares}")

# Calcular y mostrar la media aritmética de los números impares
if conteo_impares > 0:
    media_impares = suma_impares / conteo_impares
    print(f"Media aritmetica de los numeros impares: {media_impares:.2f}")
else:
    print("No hay numeros impares para calcular la media.")

File: Tarea No. 4/C++/suma_de_los_numero_imapres_en_la_lista.cpp

//Dados diez números enteros, visualizar la suma de los números pares de la lista,
//cuántos números pares existen y cuál es la media aritmética de los números impares

#include <iostream>
using namespace std;

int main() {
    int numeros[10];
    int sumaPares = 0;
    int conteoPares = 0;
    int sumaImpares = 0;
    int conteoImpares = 0;

    // Leer los 10 números enteros
    cout << "Introduce 10 numeros enteros:" << std::endl;
    for (int i = 0; i < 10; ++i) {
        cin >> numeros[i];
    }

    // Procesar los números para encontrar pares e impares
    for (int i = 0; i < 10; ++i) {
        if (numeros[i] % 2 == 0) { // Verifica si el número es par
            sumaPares += numeros[i];
            ++conteoPares;
        } else { // El número es impar
            sumaImpares += numeros[i];
            ++conteoImpares;
        }
    }

    // Mostrar resultados
    cout << "Suma de los numeros pares: " << sumaPares << endl;
    cout << "Cantidad de numeros pares: " << conteoPares << endl;

    // Calcular y mostrar la media aritmética de los números impares
    if (conteoImpares > 0) {
        double mediaImpares = static_cast<double>(sumaImpares) / conteoImpares;
        cout << "Media aritmetica de los numeros impares: " << mediaImpares << endl;
    } else {
        cout << "No hay numeros impares para calcular la media." << endl;
    }

    return 0;
}
